### ISSUE TITLE
support UNIX Domain socket as endpoint

### ISSUE THREAD

**Issue by jonassmedegaard** [joepio/atomic-data-rust#330]
When hosting via a frontend proxy sever, connecting to backend service is relatively slow to do over a TCP socket.  It is faster to use a [Unix domain socket](https://en.wikipedia.org/wiki/Unix_domain_socket) (about 6 times faster according to some measures).  Also, if frontend server does authentication, then that can be bypassed for processes running on same host.  And it can be cumbersome to keep track of which service is occupying which TCP ports.

It seems actix-web supports [listening on UNIX Domain socket](https://docs.rs/actix-web/3.3.3/actix_web/struct.HttpServer.html#method.listen_uds) since [v1.0.8](https://github.com/actix/actix-web/blob/master/actix-web/CHANGES.md#108---2019-09-25).

Please implement support for domain socket, e.g. with command-line option `--path-uds` and environment variable `ATOMIC_PATH` (as equivalent to existing `--port` and `ATOMIC_PORT`, with space to equivalently extend later to support other types of path-based socket like [uwsgi](https://uwsgi-docs.readthedocs.io/en/latest/Protocol.html)).

**Comment by joepio** [comment]
Makes sense and seems reasonable! I'm pretty sure this will be a small change. I'll look into it.
